terragrunt/aws/network/security-groups.tf

```diff
@@ -0,0 +1,61 @@
+# 
+# Security Groups for the VPC URL shortener
+#
+
+resource "aws_security_group" "api" {
+
+  name        = "${var.product_name}_api_sg"
+  description = "SG for the URL shortener API"
+
+  vpc_id = module.url_shortener_vpc.vpc_id
+
+  tags = {
+    CostCentre = var.billing_code
+    Name       = "${var.product_name}_api_sg"
+  }
+
+  egress {
+    description = "Allow API outbound connections to the internet"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "TCP"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```

Comment:
Coud you change this to a [standalone SG rule](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule).  There's nothing wrong with how you've done it as an inline block, but Terraform has trouble managing this type of rule (something we've learned the hard way on our other projects 😅)

```hcl
resource "aws_security_group_rule" "api_egress_internet" {
  description              = "API egress to the internet on 443"
  type                     = "egress"
  from_port                = 443
  to_port                  = 443
  protocol                 = "tcp"
  cidr_blocks              = ["0.0.0.0/0"]
  security_group_id        = aws_security_group.api.id
}
```

Reply:
Done!